lib/rubocop/cop/performance/zip_for_array_wrapping.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # Checks for `.map { |id| [id] }` and suggests replacing it with `.zip`.
+      #
+      # @example
+      #   # bad
+      #   [1, 2, 3].map { |id| [id] }
+      #
+      #   # bad
+      #   [1, 2, 3].map { [_1] }
+      #
+      #   # good
+      #   [1, 2, 3].zip
+      #
+      # @example
+      #   # good (no offense)
+      #   [1, 2, 3].map { |id| id }
+      #
+      # @example
+      #   # good (no offense)
+      #   [1, 2, 3].map { |id| [id, id] }
+      class ZipForArrayWrapping < Base
+        extend AutoCorrector
+
+        MSG = 'Use `zip` instead of `%<original_code>s`.'
+        RESTRICT_ON_SEND = Set.new(%i[map collect]).freeze
+
+        # @!method map_with_array?(node)
+        def_node_matcher :map_with_array?, <<~PATTERN
+          {
+            (block (send _ RESTRICT_ON_SEND) (args (arg _id)) (array (lvar _id)))
+            (numblock (send _ RESTRICT_ON_SEND) 1 (array (lvar _)))
+          }
+        PATTERN
+
+        def on_send(node)
+          return unless (parent = node.parent)
+          return unless map_with_array?(parent)
+          return unless (receiver = node.receiver&.source)
+
+          register_offense(parent, receiver, node)
+        end
```

Comment:
```suggestion
        # @!method map_with_array?(node)
        def_node_matcher :map_with_array?, <<~PATTERN
          {
            (block (call _ RESTRICT_ON_SEND) (args (arg _id)) (array (lvar _id)))
            (numblock (call _ RESTRICT_ON_SEND) 1 (array (lvar _)))
          }
        PATTERN

        def on_send(node)
          return unless (parent = node.parent)
          return unless map_with_array?(parent)
          return unless (receiver = node.receiver&.source)

          register_offense(parent, receiver, node)
        end
        alias on_csend on_send
```

I think we should also have this apply for safe navigation too please (ie. `arr&.map { _1 }`). Could you please add tests for this as well?

Reply:
Implemented!

We had to add this to the node pattern as well to get it to work.